modules/newspassidBidAdapter.js

```diff
@@ -220,10 +221,11 @@ export const spec = {
     if (getParams.hasOwnProperty('nprp') && getParams.nprp.match(/^[0-3]$/)) { extObj['newspassid']['nprp'] = parseInt(getParams.nprp); }
     if (getParams.hasOwnProperty('npip') && getParams.npip.match(/^\d+$/)) { extObj['newspassid']['npip'] = parseInt(getParams.npip); }
     if (this.propertyBag.endpointOverride != null) { extObj['newspassid']['origin'] = this.propertyBag.endpointOverride; }
-    var userExtEids = this.generateEids(validBidRequests); // generate the UserIDs in the correct format for UserId module
+    let userExtEids = deepAccess(validBidRequests, '0.userIdAsEids', []); // generate the UserIDs in the correct format for UserId module
+    let refererInfo = getRefererInfo();
     npRequest.site = {
       'publisher': {'id': htmlParams.publisherId},
-      'page': document.location.href,
+      'page': refererInfo.location,
```

Comment:
refererInfo.page i think is what you're looking for?

Reply:
updated in latest commit